pkg/operator/ceph/object/objectstore.go

```diff
@@ -88,18 +98,91 @@ func deleteRealmAndPools(context *Context, spec cephv1.ObjectStoreSpec) error {
 	return nil
 }
 
+func checkZoneIsMaster(context *Context, realmArg string, zoneGroupArg string, zoneArg string) (bool, error) {
+	zoneGroupOutput, err := RunAdminCommandNoRealm(context, "zonegroup", "get", realmArg, zoneGroupArg)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to get rgw zone group")
+	}
+	zoneGroupJson, err := decodeZoneGroupConfig(zoneGroupOutput)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to parse master zone id")
+	}
+
+	zoneOutput, err := RunAdminCommandNoRealm(context, "zone", "get", realmArg, zoneGroupArg, zoneArg)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to get rgw zone")
+	}
+	zoneID, err := decodeID(zoneOutput)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to parse zone id")
+	}
+
+	if zoneID == zoneGroupJson.MasterZoneID {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func getZoneEndpoints(context *Context, realmArg string, zoneGroupArg string, zoneName string, serviceEndpoint string) (string, error) {
+	zoneGroupOutput, err := RunAdminCommandNoRealm(context, "zonegroup", "get", realmArg, zoneGroupArg)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to get rgw zone group")
+	}
+	zoneGroupJson, err := decodeZoneGroupConfig(zoneGroupOutput)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to parse zones list")
+	}
+
+	zoneEndpoints := []string{}
+	for _, zone := range zoneGroupJson.Zones {
+		if zone.Name == zoneName {
+			for _, endpoint := range zone.Endpoints {
+				// in case object-store operator code is rereconciled, zone modify could get run again with serviceEndpoint added again
+				if endpoint != serviceEndpoint {
+					zoneEndpoints = append(zoneEndpoints, endpoint)
+				}
+			}
+			break
+		}
+	}
+
+	zoneEndpoints = append(zoneEndpoints, serviceEndpoint)
+
+	return strings.Join(zoneEndpoints, ","), nil
```

Comment:
Why not return a `[]string{}`?

Reply:
@travisn @leseb I'm not sure I follow. At the end of the day I want a string in the format `http://10.2.1.4:80,http://10.4.3.5:80,http://10.3.2.120:80`. `zoneEndpoints` is a `[]string` and so using .Join gets me the result I want. Are you suggesting I just return `zoneEndpoints` and do the .Join after this function is called?

Reply:
Correct, the suggestion is to return the slice here and `Join()` by the caller.